### ISSUE TITLE
Print helpful message when remote builder key has password

### ISSUE THREAD

**Issue by cstrahan** [NixOS/nix#1879]
I forgot that the SSH key I was using for a remote builder host had a password, and I was presented with this message:

> unable to open SSH connection to 'ssh://cstrahan@aarch64.nixos.community': cannot connect to 'cstrahan@aarch64.nixos.community'; trying other available machines...

It would be nice if the failure message included more info (in this case, that the SSH key is password protected and thus can't be used).

**Comment by vcunat** [comment]
Yes, I think people tend to run into multiple issues with setting up remote builds, and it would be helpful if the message said a bit more about the reason.

**Comment by timokau** [comment]
Is there a reason nix doesn't just ask for the password, or is it just that nobody implemented it?

**Comment by dtzWill** [comment]
> Is there a reason nix doesn't just ask for the password, or is it just that nobody implemented it?

Nix often does not have an interactive prompt where this would make sense--for example, when used by Hydra.

I'm not sure we could reasonably give a better error message here without closer integration with ssh which would take a significant amount of work (at least I'd guess this would be the case).

I suppose we may be able to save the stderr from invoking SSH and plumb it back as the error cause, assuming it's of a reasonable size?

**Comment by tomberek** [comment]
Closing as stale. Re-open if needed.
